src/ripple/overlay/impl/ProtocolMessage.h

```diff
@@ -106,80 +155,98 @@ std::pair <std::size_t, boost::system::error_code>
-    case protocol::mtLEDGER_DATA:           ec = detail::invoke<protocol::TMLedgerData> (type, buffers, handler); break;
-    case protocol::mtPROPOSE_LEDGER:        ec = detail::invoke<protocol::TMProposeSet> (type, buffers, handler); break;
-    case protocol::mtSTATUS_CHANGE:         ec = detail::invoke<protocol::TMStatusChange> (type, buffers, handler); break;
-    case protocol::mtHAVE_SET:              ec = detail::invoke<protocol::TMHaveTransactionSet> (type, buffers, handler); break;
-    case protocol::mtVALIDATION:            ec = detail::invoke<protocol::TMValidation> (type, buffers, handler); break;
-    case protocol::mtGET_OBJECTS:           ec = detail::invoke<protocol::TMGetObjectByHash> (type, buffers, handler); break;
+    case protocol::mtMANIFESTS:
+        success = detail::invoke<protocol::TMManifests>(*header, buffers, handler);
+        break;
+    case protocol::mtPING:
+        success = detail::invoke<protocol::TMPing>(*header, buffers, handler);
+        break;
+    case protocol::mtCLUSTER:
+        success = detail::invoke<protocol::TMCluster>(*header, buffers, handler);
+        break;
+    case protocol::mtGET_SHARD_INFO:
+        success = detail::invoke<protocol::TMGetShardInfo>(*header, buffers, handler);
+        break;
+    case protocol::mtSHARD_INFO:
+        success = detail::invoke<protocol::TMShardInfo>(*header, buffers, handler);
+        break;
+    case protocol::mtGET_PEER_SHARD_INFO:
+        success = detail::invoke<protocol::TMGetPeerShardInfo>(*header, buffers, handler);
+        break;
+    case protocol::mtPEER_SHARD_INFO:
+        success = detail::invoke<protocol::TMPeerShardInfo>(*header, buffers, handler);
+        break;
+    case protocol::mtENDPOINTS:
+        success = detail::invoke<protocol::TMEndpoints>(*header, buffers, handler);
+        break;
+    case protocol::mtTRANSACTION:
+        success = detail::invoke<protocol::TMTransaction>(*header, buffers, handler);
+        break;
+    case protocol::mtGET_LEDGER:
+        success = detail::invoke<protocol::TMGetLedger>(*header, buffers, handler);
+        break;
+    case protocol::mtLEDGER_DATA:
+        success = detail::invoke<protocol::TMLedgerData>(*header, buffers, handler);
+        break;
+    case protocol::mtPROPOSE_LEDGER:
+        success = detail::invoke<protocol::TMProposeSet>(*header, buffers, handler);
+        break;
+    case protocol::mtSTATUS_CHANGE:
+        success = detail::invoke<protocol::TMStatusChange>(*header, buffers, handler);
+        break;
+    case protocol::mtHAVE_SET:
+        success = detail::invoke<protocol::TMHaveTransactionSet>(*header, buffers, handler);
+        break;
+    case protocol::mtVALIDATION:
+        success = detail::invoke<protocol::TMValidation>(*header, buffers, handler);
+        break;
+    case protocol::mtGET_OBJECTS:
+        success = detail::invoke<protocol::TMGetObjectByHash>(*header, buffers, handler);
+        break;
```

Comment:
I noticed that in rewriting this `switch`, you got rid of three deprecated message types: `mtHELLO`, `mtGET_PEERS`, and `mtPEERS`. While you're at it, why not go all the way and remove those entries from the enum and all instances in other places, too? At least on mainnet, any node running a version old enough to send these is surely going to be amendment locked by now, *and* the `handler.onMessageUnknown` doesn't do anything anyway, so even on other networks, it won't hurt anything.

Reply:
Hmm, I thought I did. I'll double-check.

Reply:
Removed some leftover instances of `mtGET_PEERS` and `mtPEERS` in response this comment.